Remove timeout dependent tests in Closure tests.

qnikst opened the pull request.

qnikst commented:
@facundominguez ping?

mboes commented:
@qnikst please fix the typos in the commit message and update the PR description.

facundominguez commented on src/Control/Distributed/Process/Tests/Closure.hs:
```diff
@@ -363,19 +378,23 @@ testSpawnSupervised TestTransport{..} rtable = do
       us <- getSelfPid
       liftIO $ putMVar superPid us
-      (child, _ref) <- spawnSupervised (localNodeId node2) (waitClosure 1000000)
+      (child, _ref) <- spawnSupervised (localNodeId node2) (sendPidClosure us `seqCP` staticClosure expectClosure)
```
Could be `($mkStaticClosure 'expectUnit)`.

facundominguez replied:
Isn't this line too long?

qnikst replied:
I'm not sure about the line length in a file that don't have a strong policy about what it should be,
I've split this line into two, but didn't try to make it <80 chars.

facundominguez commented on src/Control/Distributed/Process/Tests/Closure.hs:
```diff
@@ -363,19 +378,23 @@ testSpawnSupervised TestTransport{..} rtable = do
     forkProcess node2 $ do
       [super, child] <- liftIO $ mapM readMVar [superPid, childPid]
       ref <- monitor child
+      liftIO $ putMVar linkUp ()
```
`monitor ` is asynchronous. I don't think this will work.

facundominguez commented on src/Control/Distributed/Process/Tests/Closure.hs:
```diff
@@ -363,19 +378,23 @@ testSpawnSupervised TestTransport{..} rtable = do
         putMVar childPid child
-        threadDelay 500000 -- Give the child a chance to link to us
+        liftIO $ takeMVar linkUp
+        liftIO $ putStrLn "here"
```
Print something more meaningful or remove the output.

qnikst replied:
oups.. debug output, thanks.

facundominguez commented on src/Control/Distributed/Process/Tests/Closure.hs:
```diff
@@ -363,19 +378,23 @@ testSpawnSupervised TestTransport{..} rtable = do
       liftIO $ do
         putMVar childPid child
-        threadDelay 500000 -- Give the child a chance to link to us
+        liftIO $ takeMVar linkUp
```
Remove `threadDelay` but preserve or rephrase the comment.

qnikst commented:
@facundominguez, @mboes I have addressed your comments.

@facundominguez please check if new approach to giving process a change to link is sane enough, or it uses assumptions that could not hold?

facundominguez commented:
Are the three commits in this PR intended?

qnikst commented:
Other commits is a PR that fixes test to new API, they will go away after merge of that PR.

facundominguez commented on src/Control/Distributed/Process/Tests/Closure.hs:
```diff
@@ -363,19 +378,26 @@ testSpawnSupervised TestTransport{..} rtable = do
       ref <- monitor child
+      -- because monitor message was sent before message to process
+      -- we hope that it will be processed before
+      send p ()
```
Monitor sends a message to the local NC:
```Haskell
monitor' :: Identifier -> Process MonitorRef
monitor' ident = do
  monitorRef <- getMonitorRefFor ident
  sendCtrlMsg Nothing $ Monitor monitorRef
  return monitorRef
```
I don't think you are earning any ordering guarantees.

There is one other place in tests where waiting for monitoring is [done](https://github.com/haskell-distributed/distributed-process-tests/blob/5e426233313a532c237f2b22edce5d8d1fa1b916/src/Control/Distributed/Process/Tests/CH.hs#L120).

facundominguez replied:
I think Tim expected this to be doable with the management interface but the monitoring event was not yet implemented.

facundominguez commented on src/Control/Distributed/Process/Tests/Closure.hs:
```diff
@@ -141,11 +145,22 @@ factorial' n = returnCP $(mkStatic 'sdictInt) n `bindCP` factorialOf
 
-simulateNetworkFailure :: TestTransport -> NodeId -> NodeId -> Process ()
+expectClosure :: Static (Process ())
+expectClosure = $(mkStatic 'expectUnit)
```
Unused.

qnikst commented:
I've addressed comments, PTAL.

facundominguez commented on src/Control/Distributed/Process/Tests/Closure.hs:
```diff
@@ -363,19 +376,35 @@ testSpawnSupervised TestTransport{..} rtable = do
+    linkUp               <- newEmptyMVar
 
-    forkProcess node1 $ do
+    p <- forkProcess node1 $ do
```
`p` seems unused.

facundominguez commented:
:+1:
